Brokerages/Alpaca/AlpacaBrokerageFactory.cs

```diff
@@ -0,0 +1,108 @@
+        /// Initializes a new instance of the <see cref="AlpacaBrokerageFactory"/> class.
+        /// </summary>
+        public AlpacaBrokerageFactory()
+            : base(typeof(AlpacaBrokerage))
+        {
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public override void Dispose()
+        {
+        }
+
+        /// <summary>
+        /// Gets the brokerage data required to run the brokerage from configuration/disk
+        /// </summary>
+        /// <remarks>
+        /// The implementation of this property will create the brokerage data dictionary required for
+        /// running live jobs. See <see cref="IJobQueueHandler.NextJob"/>
+        /// </remarks>
+        public override Dictionary<string, string> BrokerageData
+        {
+            get
+            {
+                return new Dictionary<string, string>
+                {
+                    { "alpaca-key-id", Config.Get("alpaca-key-id") },
+                    { "alpaca-secret-key", Config.Get("alpaca-secret-key") },
+                    { "alpaca-base-url", Config.Get("alpaca-base-url") }
+                };
+            }
+        }
+
+        /// <summary>
+        /// Gets a new instance of the <see cref="AlpacaBrokerageModel"/>
+        /// </summary>
+        public override IBrokerageModel BrokerageModel
+        {
+            get { return new AlpacaBrokerageModel(); }
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="IBrokerage"/> instance
+        /// </summary>
+        /// <param name="job">The job packet to create the brokerage for</param>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <returns>A new brokerage instance</returns>
+        public override IBrokerage CreateBrokerage(LiveNodePacket job, IAlgorithm algorithm)
+        {
+            var errors = new List<string>();
+
+            // read values from the brokerage data
+            var keyId = Read<string>(job.BrokerageData, "alpaca-key-id", errors);
+            var secretKey = Read<string>(job.BrokerageData, "alpaca-secret-key", errors);
+            var tradingMode = Read<string>(job.BrokerageData, "alpaca-trading-mode", errors);
+
+            if (errors.Count != 0)
+            {
+                // if we had errors then we can't create the instance
+                throw new Exception(string.Join(System.Environment.NewLine, errors));
+            }
+
+            tradingMode = tradingMode.ToLower();
+            if (!tradingMode.Equals("live") && !tradingMode.Equals("paper"))
+            {
+                // if the trading mode is invalid, do not proceed further
+                throw new Exception("Available trading mode: paper/live");
+            }
+
+            var brokerage = new AlpacaBrokerage(algorithm.Transactions, algorithm.Portfolio, keyId, secretKey, tradingMode);
```

Comment:
Can you put the trading mode into the `BrokerageData` parameters and remove the base url? Given the baseUrl is hard coded in the constructor it looks like it shouldn't be in `BrokerageData`.

Reply:
I guess I've put the trading mode into the BrokerageData and removed the baseUrl from it.

Did I do sth wrong?

Please let me know.  Thanks.